auth/services/irma/signer.go

```diff
@@ -138,6 +144,35 @@ func (v Service) SigningSessionStatus(sessionID string) (contract.SigningSession
 	return nil, services.ErrSessionNotFound
 }
 
+func (v Service) legalEntityFromContract(sic *SignedIrmaContract) (*did.DID, error) {
+	params := sic.contract.Params
+
+	q, err := v.VCResolver.Registry().QueryFor(concept.OrganizationConcept)
+	if err != nil {
+		return nil, err
+	}
+	q.AddClause(concept.Eq(concept.OrganizationName, params[contract.LegalEntityAttr]))
+	q.AddClause(concept.Eq(concept.OrganizationCity, params[contract.LegalEntityCityAttr]))
+
+	c, err := v.VCResolver.Search(q)
+	if err != nil {
+		return nil, err
+	}
+	if len(c) == 0 {
+		return nil, ErrLegalEntityNotFound
+	}
+
+	cs := make([]credential.BaseCredentialSubject, 0)
```

Comment:
I'd handle the case when there's multiple matches (= error) since it either means someone made a mistake or there's a deployment scenario we haven't anticipated (multiple vendors having the same care organizations). Either way I think it's better to fail loudly rather than silently do the (possibly) wrong thing.

Furthermore when there's multiple matches for the same DID (multiple VCs with the same subject to the same DID) they can be considered equal. Although that's such an edge case we might not want to support (yet).

Reply:
Multiple DIDs with the same name/city would require a rewrite of the session handling part. I'll add an issue for that: the DID will have to be stored or passed in the request so the `legalEntityFromContract` can match against the DID and not search it. #196

Reply:
Then shouldn't we for now fail if there's multiple matches instead of picking a random one?

Reply:
I think it would hinder us now in dev/test